### PR TITLE
lockups: use expectedErr with error types

### PR DESCRIPTION
**Pull request by RusAkh**
Closes partially: #2959

What is the purpose of the change:
Update tests that are using expectPass or expectErr as the bool variable to the error type in the lockup module

all tests pass

### CODE REVIEW

File: x/lockup/types/errors.go [hunk 1]
```diff
@@ -12,4 +12,9 @@ var (
 	ErrSyntheticLockupAlreadyExists      = sdkerrors.Register(ModuleName, 2, "synthetic lockup already exists for same lock and suffix")
 	ErrSyntheticDurationLongerThanNative = sdkerrors.Register(ModuleName, 3, "synthetic lockup duration should be shorter than native lockup duration")
 	ErrLockupNotFound                    = sdkerrors.Register(ModuleName, 4, "lockup not found")
+	ErrNewDurationNotGreater             = sdkerrors.Register(ModuleName, 5, "new duration should be greater than the original")
+	ErrRequestedExceedsTokens            = sdkerrors.Register(ModuleName, 6, "requested amount to unlock exceeds locked tokens")
+	ErrNotStartedUnlocking               = sdkerrors.Register(ModuleName, 7, "lock hasn't started unlocking yet")
+	ErrNotUnlockableYet                  = sdkerrors.Register(ModuleName, 8, "lock is not unlockable yet")
+	ErrSyntheticLockupNotExists          = sdkerrors.Register(ModuleName, 9, "synthetic lockup does not exist")
```
Comment (mattverse) [inline x/lockup/types/errors.go:19]: I think these should be handled in a separate PR + needs further discussion on adding these, as adding new errors would make this PR state breaking

Comment (RusAkh) [reply x/lockup/types/errors.go:19]: I can remove them for now but the tests would be pretty dirty without those, what do you think? @mattverse

Comment (czarcas7ic) [reply x/lockup/types/errors.go:19]: Since this PR is not urgent, we can probably just leave these here and not backport it right?

**Comment by czarcas7ic** [comment]
Going to close this for now and we can potentially reopen after further discussions
